node-fs-readfile/cat.js

```diff
@@ -0,0 +1,22 @@
+import { readFile } from 'node:fs/promises';
+
+async function readCatAnyFiles() {
+  const fileArray = process.argv.slice(2);
+
+  const promiseArray = fileArray.map(async (file) => {
+    const filePath = new URL(file, import.meta.url);
+    const contents = await readFile(filePath, { encoding: 'utf8' });
+    return contents;
+  });
+
+  const fileContents = await Promise.all(promiseArray);
+  fileContents.forEach((fileContent) => {
+    console.log(fileContent);
+  });
+}
+
+try {
+  await readCatAnyFiles();
+} catch (error) {
+  console.error('Error:', error.message);
```

Comment:
You usually want to log the entire error, not just the message field. That way you will get a stack trace that will help you figure out where things went wrong.

Reply:
Updated